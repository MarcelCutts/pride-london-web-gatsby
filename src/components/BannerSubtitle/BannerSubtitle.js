import React from 'react'
import styled from 'styled-components'

const BannerSubtitle = styled.h3`
  color: #ffffff;
  font-size: 16px;

  & span {
    background-color: #2d2f7f;
    display: inline-block;
    padding: 0 10px;
  }
`
const StyledBannerSubtitle = ({ children }) => (
  <BannerSubtitle>
    <span>{children}</span>
  </BannerSubtitle>
)

export default StyledBannerSubtitle
