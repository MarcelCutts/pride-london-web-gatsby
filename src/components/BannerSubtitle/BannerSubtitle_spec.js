import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { shallow } from 'enzyme'

import BannerSubtitle from '../BannerSubtitle/BannerSubtitle'

describe('BannerSubtitle', () => {
  it('should render', () => {
    const tree = renderer.create(<BannerSubtitle />)
    expect(tree).toMatchSnapshot()
  })

  it('should render a <span />', () => {
    const wrapper = shallow(<BannerSubtitle />)
    expect(wrapper.find('span')).toHaveLength(1)
  })

  it('should render child text ', () => {
    const text = 'And here is a test subtitle!'
    const wrapper = shallow(<BannerSubtitle>{text}</BannerSubtitle>)
    expect(wrapper.find('span').text()).toBe(text)
  })
})
