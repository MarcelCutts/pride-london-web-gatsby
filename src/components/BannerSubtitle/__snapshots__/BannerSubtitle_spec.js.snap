// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerSubtitle should render 1`] = `
.c0 {
  background-color: #2d2f7f;
  color: #ffffff;
  display: inline-block;
  font-size: 16px;
  padding: 0 10px;
}

<h3
  className="c0"
>
  <span />
</h3>
`;
